src/core.jl

```diff
@@ -6,6 +6,28 @@
 ### statelength(l)            --> Integer
 ### l(x, ps, st)
 
+"""
+    AbstractExplicitLayer
+
+Abstract Type for all Lux Layers
+
+Users implementing their custom layer, **must** implement
+
+  - `initialparameters(rng::AbstractRNG, layer::CustomAbstractExplicitLayer)` -- This
+    returns a `NamedTuple` containing the trainable parameters for the layer.
+  - `initialstates(rng::AbstractRNG, layer::CustomAbstractExplicitLayer)` -- This returns a
```

Comment:
My understanding is that if my custom layer does not have a state then I don't need to implement this function? Since it is calling `initialstates(::AbstractRNG, ::Any) = NamedTuple()`